### ISSUE TITLE
Unrecognised configuration name: site_favicon

### ISSUE THREAD

**Issue by kcgthb** [squidfunk/mkdocs-material#551]
## Description

MkDocs 0.17.x generates a warning on `site_favicon`:
```
WARNING -  Config value: 'site_favicon'. Warning: Unrecognised configuration name: site_favicon
```
which makes build fail when using `--strict`.

### Expected behavior

No warning, so `mkdocs build --strict` could succeed.

### Actual behavior

```
$ mkdocs build --strict
WARNING -  Config value: 'site_favicon'. Warning: Unrecognised configuration name: site_favicon

Aborted with 1 Configuration Warnings in 'strict' mode!
```

### Steps to reproduce the bug

1. Upgrade to MkDocs 0.17.1
2. Upgrade to mkdocs-material 2.0.1
3. use `site_favicon` in `mkdocs.yml`

### Package versions

* Python: `python --version`: 3.4.5
* MkDocs: `mkdocs --version`: 0.17.1
* Material: `pip show mkdocs-material | grep -E ^Version`: 2.0.1

### Project configuration

``` yaml
# Site information
site_name:        Test
[...]
site_favicon:     img/logo_small.png
[...]
```

### System information

* OS: Linux
* Browser: Chrome

**Comment by martinbira** [comment]
As per the MkDocs documentation on http://www.mkdocs.org/#changing-the-favicon-icon

> By default, MkDocs uses the MkDocs favicon icon. To use a different icon, create an `img` subdirectory in your `docs_dir` and copy your custom `favicon.ico` file to that directory. MkDocs will automatically detect and use that file as your favicon icon.

So remove the site_favicon from `mkdocs.yml` and rename the file to `favicon.ico` and things should work again.

**Comment by squidfunk** [comment]
Well, that doesn't work if you want to use a PNG file or other image type as a favicon, which all major browsers support by now. I'll look into that.

**Comment by squidfunk** [comment]
I'll move it into the theme configuration, as MkDocs now doesn't support configuring it.

**Comment by squidfunk** [comment]
Fixed in #555

**Comment by vsoch** [comment]
Just ran into this question! Putting the icon in the img folder worked for me. But for future reference (e.g., custom extension need), here is the pretty rendered answer:

https://squidfunk.github.io/mkdocs-material/getting-started/#favicon
